### PR TITLE
Feat heatmap

### PR DESCRIPTION
**Pull request by philipkueng**
Deployed on st-staging
https://q.st-staging.nzz.ch/item/table-36
https://q.st-staging.nzz.ch/item/table-41

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -283,6 +279,108 @@ Minibars are a visual feature to display the difference between numbers in the t
...
+  categoricalOptions, // object
+  colorColumnType, // string
+  colors, // array
+  legendData, // object
+  numericalOptions, // object
+  selectedColumn, // number
+}
+```
+
+- Depedning on the selected `colorColumnType`, either `numerical` or `categorical`, some of the properties will be added
```
Comment (manuelroth) [inline README.md:367]:
```suggestion
- Depending on the selected `colorColumnType`, either `numerical` or `categorical`, some of the properties will be added 
```

File: README.md [hunk 2]
```diff
@@ -283,6 +279,108 @@ Minibars are a visual feature to display the difference between numbers in the t
...
+```javascript
+{
+  methodBox, // object
+  formattedValue, // array 
+}
+```
+
+###### Implementation details frontend
+
+- Either `numerical` or `categorical`, both table-cells will be colored with the referece to the `colors` array
```
Comment (manuelroth) [inline README.md:379]:
```suggestion
- Either `numerical` or `categorical`, both table-cells will be colored with the reference to the `colors` array
```

**Review by fromdusttilldawn — COMMENTED** [review]
The Methodik-Box doesn't seem to work. When I click on "Daten und Methodik" nothing happens.

File: resources/fixtures/data/colorColumn-categorical.json [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          "colorCode": ""
+        },
+        "negative": {
+          "className": "",
+          "colorCode": ""
+        }
+      },
+      "selectedColumn": null
+    },
+    "colorColumn": {
```
Comment (manuelroth) [inline resources/fixtures/data/colorColumn-categorical.json:72]: @philipkueng do `colorColumn` options have to be present for the tool to work?- Or are they optional. Do you have to do a migration to roll this feature out?

File: resources/locales/de/translation.json [hunk 4]
```diff
@@ -1 +1,20 @@
...
+            "title": "Manuelle Grenzen",
+            "body": "Die manuellen Grenzen umfassen nicht alle der in der Datentabelle eingegebenen Werte."
+        },
+        "numberBucketsExceedsDataSet": {
+            "title": "Anzahl Buckets",
+            "body": "Die Anzahl Buckets ist grösser als die Anzahl unterschiedlicher Werte in der Datentabelle"
+        },
+        "numberCategoriesOutOfColorScale": {
+            "title": "Farbskala",
+            "body": "Die Anzahl unterschiedlicher Kategorien ist höher als wir mit unserer Hauptfarbskala abdecken können"
```
Comment (manuelroth) [inline resources/locales/de/translation.json:13]:
```suggestion
            "body": "Die Anzahl unterschiedlicher Kategorien ist höher als mit der Hauptfarbskala abdeckt werden kann"
```
I would formulate notification messages always in a general way.

File: resources/locales/de/translation.json [hunk 5]
```diff
@@ -1 +1,20 @@
...
+        },
+        "numberBucketsOutOfColorScale": {
+            "title": "Farbskala",
+            "body": "Die Anzahl Buckets ist höher als wir mit unseren Farbskalen abdecken können"
```
Comment (manuelroth) [inline resources/locales/de/translation.json:17]:
```suggestion
            "body": "Die Anzahl Buckets ist höher als mit den Farbskalen abgedeckt werden kann."
```

File: routes/option-availability.js [hunk 6]
```diff
@@ -47,79 +50,133 @@ module.exports = {
...
+        };
+      }
+
+      if (optionName === "isCategorical") {
+        return {
+          available: item.options.colorColumn.colorColumnType === "categorical",
+        };
+      }
+
+      if (optionName === "colorColumnType" || optionName === "bucketType" || optionName === "scale" || optionName === "colorOverwritesItem" || optionName === "colorScheme" || optionName === "customCategoriesOrder") {
```
Comment (manuelroth) [inline routes/option-availability.js:143]:
```suggestion
      if (["colorColumnType", "bucketType", "scale", "colorOverwritesItem", "colorScheme", "customCategoriesOrder"].includes(optionName)) {
```

**Comment by manuelroth** [comment]
This is a very big feature. I think it turned out great. Well done @philipkueng. There is a lot of logic in helper methods, but still I think the nunjucks-templates are super complex. I had a very hard time understanding what they are doing. I think we should plan a project to convert the tool to svelte, before we plan any new features for this tool in the future.

**Comment by manuelroth** [comment]
This feature 

> The Methodik-Box doesn't seem to work. When I click on "Daten und Methodik" nothing happens.

I also noticed that. @philipkueng I created [a basecamp todo](https://3.basecamp.com/3500782/buckets/1333707/todos/3712174060) for this as well

**Review by manuelroth — APPROVED** [review]

**Pull request merged**